timemachine/potentials/nonbonded.py

```diff
@@ -155,399 +155,43 @@ def nonbonded_v3(
 
     eps_ij = eps_i * eps_j
 
-    ri = np.expand_dims(conf, 0)
-    rj = np.expand_dims(conf, 1)
-    dij = distance(ri, rj, box)
+    dij = distance(conf, box)
 
     N = conf.shape[0]
     keep_mask = np.ones((N,N)) - np.eye(N)
     keep_mask = np.where(eps_ij != 0, keep_mask, 0)
 
     if cutoff is not None:
-        eps_ij = np.where(dij < cutoff, eps_ij, np.zeros_like(eps_ij))
+        eps_ij = np.where(dij < cutoff, eps_ij, 0)
 
     # (ytz): this avoids a nan in the gradient in both jax and tensorflow
-    sig_ij = np.where(keep_mask, sig_ij, np.zeros_like(sig_ij))
-    eps_ij = np.where(keep_mask, eps_ij, np.zeros_like(eps_ij))
+    sig_ij = np.where(keep_mask, sig_ij, 0)
+    eps_ij = np.where(keep_mask, eps_ij, 0)
 
-    sig2 = sig_ij/dij
+    inv_dij = 1/dij
+    inv_dij = np.where(np.eye(N), 0, inv_dij)
+
+    sig2 = sig_ij*inv_dij
     sig2 *= sig2
     sig6 = sig2*sig2*sig2
 
     eij_lj = 4*eps_ij*(sig6-1.0)*sig6
-    eij_lj = np.where(keep_mask, eij_lj, np.zeros_like(eij_lj))
+    eij_lj = np.where(keep_mask, eij_lj, 0)
 
     qi = np.expand_dims(charges, 0) # (1, N)
     qj = np.expand_dims(charges, 1) # (N, 1)
     qij = np.multiply(qi, qj)
 
     # (ytz): trick used to avoid nans in the diagonal due to the 1/dij term.
     keep_mask = 1 - np.eye(conf.shape[0])
-    qij = np.where(keep_mask, qij, np.zeros_like(qij))
-    dij = np.where(keep_mask, dij, np.zeros_like(dij))
+    qij = np.where(keep_mask, qij, 0)
+    dij = np.where(keep_mask, dij, 0)
 
     # funny enough lim_{x->0} erfc(x)/x = 0
-    eij_charge = np.where(keep_mask, qij*erfc(beta*dij)/dij, np.zeros_like(dij)) # zero out diagonals
+    eij_charge = np.where(keep_mask, qij*erfc(beta*dij)*inv_dij, 0) # zero out diagonals
     if cutoff is not None:
-        eij_charge = np.where(dij > cutoff, np.zeros_like(eij_charge), eij_charge)
+        eij_charge = np.where(dij > cutoff, 0, eij_charge)
 
     eij_total = (eij_lj*lj_rescale_mask + eij_charge*charge_rescale_mask)
 
     return np.sum(eij_total/2)
-
```

Comment:
While fresh in mind, could you comment here on the deletion of the jax-based nonbonded functions?

(`lennard_jones`, `lennard_jones_exclusion`,  `self_energy`,  `simple_energy`, `pairwise_energy`, `reciprocal_energy`, `electrostatics`, `ewald_energy`)

Reply:
```lennard_jones, lennard_jones_exclusion, electrostatics,  simple_energy, pairwise_energy``` were older implementations of the non-fused potentials. 

```ewald_energy, reciprocal_energy,  self_energy``` were deprecated, untested Ewald implementations. At one point there were bugs with these functions since they required a derivative through complex numbers.